docs/source/download.md

```diff
@@ -0,0 +1,60 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Download
+
+The official Apache DataFusion releases are provided as source artifacts.
+
+## Releases
+
+The latest source release is [0.45.0](https://www.apache.org/dyn/closer.lua/opendal/0.45.0/apache-opendal-0.45.0-src.tar.gz?action=download) ([asc](https://downloads.apache.org/opendal/0.45.0/apache-opendal-0.45.0-src.tar.gz.asc),
```

Comment:
I assume the opendal references need replacing with datafusion?

Reply:
Thanks, I missed that. We are planning on creating a new release this week. I am going to start the process tomorrow.

https://github.com/apache/datafusion/issues/10217

Reply:
We could link to the last release in the arrow site for now?

https://downloads.apache.org/arrow/arrow-datafusion-37.0.0/